agents/gce/fence_gce.py

```diff
@@ -97,6 +206,25 @@ def main():
 
 	run_delay(options)
 
+	# Prepare logging
+	if options.get('--stackdriver-logging') is not None:
+		try:
+			import google.cloud.logging.handlers
+			client = google.cloud.logging.Client()
+			handler = google.cloud.logging.handlers.CloudLoggingHandler(client, name=hostname)
+			handler.setLevel(logging.INFO)
+			formatter = logging.Formatter('gcp:stonish "%(message)s"')
+			handler.setFormatter(formatter)
+			root_logger = logging.getLogger()
+			if options.get('--verbose') is None:
+				root_logger.setLevel(logging.INFO)
```

Comment:
This doesnt make any sense. The logic for the root_logger is taken care of in fencing.py.

Reply:
yes, but if we use `--stackdriver-logging`, the logs are not being sent to stackdriver because the logging level defaults to WARN and not INFO.

Reply:
Cant you just add the handler above (which has the correct level for stackdriver logging) like we do in the fencing library?

Example:
https://github.com/ClusterLabs/fence-agents/blob/master/lib/fencing.py.py#L723

Reply:
I am not sure I understand what you mean, I added the handler here: https://github.com/ClusterLabs/fence-agents/pull/211/files/9ae0a072424fa982e1d18a2cb661628c38601c3a#diff-4a0f3ce2f5196c812280cc52f7aa42b6R222

Reply:
Oh. I didnt find that. But doesnt that log correctly based on the handler.setLevel() above?

Reply:
no, this is just a filter to the handler. The root logger get the logs based in the root logger level, then it sends the logs to the handlers which filter the messages again with its own level, but if the logs does't get thought the root logger in the first place, then nothing is sent to the handlers.

Reply:
Ah. I guess we'll just have to live with that, and I might improve that later by lowering the other sub-loggers in this case.

Reply:
This is a nice chart flow explaining that: https://docs.python.org/3/howto/logging.html#logging-flow

Reply:
Thanks.